### PR TITLE
Add a GUI for layered fog

### LINKED ISSUE #1479: Add UI for layered fog
#1428 needs a UI, and since this would be fairly complex it's being split from the functionality and worked on separately. Here's a potential mockup:
notlabelled
Labelled:
labelled

There has been [some discussion](https://discord.com/channels/541221265512464394/1032297295795982396/1032297839776251914) on the discord about this UI, and this issue can serve as a tracker for work related to this.
Hopefully @ShirleyNekoDev can work some magic on this, and open a PR when we're closer to a working prototype.

### PR DESCRIPTION
**Pull request by Peregrine05**
Closes #1479.
Instead of creating a more complicated GUI as shown in the issue, I decided to create a simpler one.

**New:**
image
image

### CODE REVIEW

**Comment by megamasterbloc** [comment]
maybe change layer breath to a more meaningful term like layer dispersion or layer spreading

**Comment by Peregrine05** [comment]
It is called `breadth` internally, and it is an accurate term in my opinion.

File: chunky/src/java/se/llbit/chunky/renderer/scene/Fog.java [hunk 1]
```diff
@@ -48,6 +54,47 @@ public boolean fastFog() {
...
+    int currentIndex = 0;
+    int newIndex = 0;
+    for (FogLayer layer : layers) {
+      if (!(currentIndex == index)) {
+        layers2[newIndex] = layer;
+        newIndex++;
+      }
+      currentIndex++;
+    }
+    layers = Arrays.copyOf(layers2, layers2.length);
```
Comment (leMaik) [inline chunky/src/java/se/llbit/chunky/renderer/scene/Fog.java:79]: Wouldn't it be way easier to use a list?

Comment (Peregrine05) [reply chunky/src/java/se/llbit/chunky/renderer/scene/Fog.java:79]: I'll attempt the conversion. At some point, I did store it as an ArrayList, which was much easier, but I was tricked into thinking that it was causing an issue, where the issue was actually due to the minimum value of *0.01* for fog breadth, which makes fog rendering very unstable and likely to crash or hang forever. By then, I had already changed it back to Array, though.

**Comment by Peregrine05** [comment]
Should I change the name of the `Layer breadth` control, or leave it as it is?

**Comment by jackjt8** [comment]
> Should I change the name of the `Layer breadth` control, or leave it as it is?

The definition of Breadth is "the the distance from one side to another". I haven't looked at the code as to exactly what it's used for but I'm pretty sure Breadth controls the vertical spread of the fog layer. Which.. technically would mean the wrong term is use?

**Comment by Peregrine05** [comment]
I wonder if `Veritcal spread` would be a suitable replacement.

**Comment by leMaik** [comment]
How about `layer height`?

**Comment by Peregrine05** [comment]
> How about `layer height`?

That would be too easily confused with `Layer altitude`, I believe.

**Comment by Peregrine05** [comment]
I changed it to `Layer thickness`, which should be easily understandable.

**Review by leMaik — APPROVED** [review]

**Comment by leMaik** [comment]
@Peregrine05 Somehow, picking the fog color of any layer changes the color of all layers.

**Comment by Peregrine05** [comment]
> @Peregrine05 Somehow, picking the fog color of any layer changes the color of all layers.

@leMaik This is actually a limitation of the layered fog, and is mentioned in #1428.

**Comment by leMaik** [comment]
Indeed, it wasn't clear how to blend the layers iirc. Can we make it obvious in the UI that the color affects all layers? Or is it obvious and I'm just blind?

**Comment by Peregrine05** [comment]
It does not seem obvious to me. Another hint could be useful.

**Comment by megamasterbloc** [comment]
while it's a limitation of chunky I think that it's perfectly reasonable to only have one color for all fog layers. layered fog is only meant to change the distribution of fog and not it's properties

**Comment by Peregrine05** [comment]
I have added a new note / hint to the fog GUI.

**Pull request merged**